File: services/queue.service.go
package job

import (
	"context"
	"distributed-task-queue/domain"
	postgresql "distributed-task-queue/internal/repository/postgresql"
)

type QueueService struct {
	queueRepo postgresql.QueueRepository
}

func NewQueueService(queueRepo postgresql.QueueRepository) *QueueService {
	return &QueueService{queueRepo: queueRepo}
}

func (s *QueueService) CreateQueue(ctx context.Context, queue domain.Queue) (domain.Queue, error) {
	return s.queueRepo.CreateQueue(ctx, queue)
}

func (s *QueueService) GetQueue(ctx context.Context, id int64) (domain.Queue, error) {
	return s.queueRepo.GetByID(ctx, id)
}

File: src/services/queue.service.ts
import { QueueRepository } from '../repository/postgresql/queue.repository';
import { Queue, CreateQueueInput } from '../domain/queue';

export class QueueService {
  constructor(private queueRepo: QueueRepository) {}

  async createQueue(input: CreateQueueInput): Promise<Queue> {
    return this.queueRepo.createQueue(input);
  }

  async getQueue(id: number): Promise<Queue> {
    return this.queueRepo.getById(id);
  }

  async getAllQueues(): Promise<Queue[]> {
    return this.queueRepo.getAll();
  }
}

